src/vanilla/utils/atomWithDefault.ts

```diff
@@ -1,5 +1,5 @@
 import { atom } from '../../vanilla.ts'
-import type { SetStateAction, WritableAtom } from '../../vanilla.ts'
```

Comment:
Not sure if it's useful but in our codebase we defined a second generic type parameter to SetStateAction that allows a different return type from update functions but defaults to the first type parameter to let us express things like the SetStateWithReset type:

`export type SetStateAction<S, R = S> = R | ((prevState: S) => R);`

so you can say

`type SetStateActionWithReset<Value> = SetStateAction<Value, Value | typeof RESET>`

Maybe that's over optimizing if reset is the only use case for a differing return type but just thought I'd mention it :)

Reply:
Yeah, usually SetStateAction has the same type for the arg value and the return value.